tools/test/test_osbuild_image_info.py

```diff
@@ -135,3 +138,34 @@
 def test_read_boot_entries(tmp_path, fake_tree, entries):
     make_fake_tree(tmp_path, fake_tree)
     assert osbuild_image_info.read_boot_entries(tmp_path / "boot") == entries
+
+
+def make_fake_iso(iso_tree, output_dir) -> str:
+    iso_path = os.path.join(output_dir, "image.iso")
+    subprocess.run(["genisoimage", "-o", iso_path, "-R", "-J", iso_tree], check=True)
+    return iso_path
+
+
+def test_analyse_iso_fail(tmp_path):
+    # fake ISO that can't be mounted
+    image_path = tmp_path / "image.iso"
+    image_path.touch()
+    try:
+        osbuild_image_info.analyse_iso(image_path)
+    # pylint: disable=broad-exception-caught
+    except Exception as e:
+        assert str(e).startswith(f"Failed to mount ISO {image_path}:")
+    else:
+        assert False, "Expected an exception, but didn't get any"
+
+    # ISO that can be mounted, but doesn't contain the liveimg.tar.gz
```

Comment:
(nitpick) I wonder if it would be nicer to split this into two tests, the setup seems short enough (so not much duplication)?

Reply:
I actually already did it locally after adding the skip if not root, so +1 :innocent: